Assets/MRTK/SDK/Editor/Inspectors/UX/InteractiveElement/TrackedStatesInspector.cs

```diff
@@ -0,0 +1,282 @@
+                    SerializedProperty stateName = state.FindPropertyRelative("stateName");
+                    SerializedProperty stateValue = state.FindPropertyRelative("stateValue");
+                    SerializedProperty stateEventConfiguration = state.FindPropertyRelative("eventConfiguration");
+
+                    EditorGUILayout.Space();
+
+                    using (new EditorGUILayout.HorizontalScope())
+                    {
+                        InspectorUIUtility.DrawLabel(stateName.stringValue, 14, InspectorUIUtility.ColorTint10);
+
+                        // Draw the state value in the state while the application is playing for debugging
+                        if (Application.isPlaying)
+                        {
+                            EditorGUILayout.LabelField(stateValue.intValue.ToString());
+                        }
+
+                        // Do not draw a remove button for the default state
+                        if (stateName.stringValue != "Default")
+                        {
+                            // Draw a button with a '-' for state removal
+                            if (InspectorUIUtility.SmallButton(RemoveStateButtonLabel))
+                            {
+                                stateList.DeleteArrayElementAtIndex(i);
+                                break;
+                            }
+                        }
+                    }
+
+                    using (new EditorGUILayout.VerticalScope())
+                    {
+                        EditorGUILayout.Space();
+                        EditorGUILayout.Space();
+
+                        using (new EditorGUI.IndentLevelScope())
+                        {
+                            // If an event configuration for a state exists then draw the event config scriptable 
+                            RenderStateEventConfiguration(stateName, stateEventConfiguration);
+
+                            // When a new core state is added via inspector, the name is initialized to "New Core State" and then changed
+                            // to the name the user selects from the enum list of CoreInteractionStates
+                            if (stateName.stringValue == "New Core State")
+                            {
+                                SetCoreStateType(state, stateName);
+                            }
+
+                            // When a new state is added via inspector, the name is initialized to "Create New State" and then changed
+                            // to the name the user enters a name in the text field and then selects the "Set State Name" button
+                            if (stateName.stringValue == "Create New State")
+                            {
+                                using (new EditorGUILayout.HorizontalScope())
+                                {
+                                    newStateName = EditorGUILayout.TextField("New State Name", newStateName);
```

Comment:
Avoid magic values, see my comments below. In general, it would be good to do a sweep and remove any hard coded "string value names" == comparisons.

Reply:
Added constants